tools/topology/topology2/cavs/cavs-rt5682-rt1019.conf

```diff
@@ -0,0 +1,250 @@
+<include/common/vendor-token.conf>
+<include/common/tokens.conf>
+<include/common/manifest.conf>
+<include/common/pcm.conf>
+<include/common/pcm_caps.conf>
+<include/common/fe_dai.conf>
+<include/common/route.conf>
+<include/components/host.conf>
+<include/components/dai.conf>
+<include/components/pipeline.conf>
+<include/components/copier.conf>
+<include/dais/ssp.conf>
+<include/dais/dmic.conf>
+<include/dais/pdm_config.conf>
+<include/common/data.conf>
+<include/dais/hw_config.conf>
+<include/pipelines/cavs/passthrough-playback.conf>
+<include/pipelines/cavs/passthrough-capture.conf>
+
+#
+# List of all DAIs
+#
+# PCM0 ----> SSP0  (Headset - ALC5682)
+# PCM1 <---> SSP1  (Speaker - ALC1019)
+# PCM2 <---- DMIC01 (dmic 48k capture)
+Object.Dai {
+	SSP."0" {
+		id 		0
+		direction	"duplex"
+		name		SSP0-Codec
+		default_hw_conf_id	0
+		sample_bits		24
+
+		Object.Base.hw_config."SSP0" {
+			id	0
+			mclk_freq	19200000
+			bclk_freq	2400000
+			tdm_slot_width	25
+		}
+
+		# include DAI copier components
+		Object.Widget.copier."0" {
+			index 1
+			dai_index 1
+			type "dai_in"
+			dai_type "SSP"
+			copier_type "SSP"
+			direction "playback"
+			stream_name "SSP0-Codec"
+			period_sink_count 0
+			period_source_count 2
+			format s24le
+		}
+
+		Object.Widget.copier."1" {
+			index 2
+			dai_index 2
+			type "dai_out"
+			dai_type "SSP"
+			copier_type "SSP"
+			direction "capture"
+			stream_name "SSP0-Codec"
+			period_sink_count 2
+			period_source_count 0
+			format s24le
+		}
+	}
+	# sof_rt5682.c in machine driver - id for speaker is 6 - 1 headset + 2 dmic + 3 hdmi (if not defined in quirks)
+	SSP."1" {
+		id 		6
+		direction	"playback"
+		name		SSP1-Codec
+		default_hw_conf_id	0
+		sample_bits		24
+
+		Object.Base.hw_config."SSP1" {
+			id	0
+			mclk_freq	19200000
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+
+		# include DAI copier components
+		Object.Widget.copier."0" {
+			index 3
+			dai_index 1
+			type "dai_in"
+			dai_type "SSP"
+			copier_type "SSP"
+			direction "playback"
+			stream_name "SSP1-Codec"
+			period_sink_count 0
+			period_source_count 2
+			format s24le
+		}
+	}
+	DMIC."0" {
+		name dmic01
+		id 2
+		driver_version		1
+		clk_min			500000
+		clk_max			4800000
+		duty_min		40
+		duty_max		60
+		sample_rate		48000
+		fifo_word_length	16
+		unmute_ramp_time_ms	200
+		num_pdm_active 1
+
+		# these fir design parameters are for future use
+		# and they are currently ignored
+		fir_type       		 "kaiser-linear-phase"
+		fir_length		 127
+		cic_decimate		 25
+		fir_decimate		 2
+		fir_passband_end	 20000
+		fir_stopband_start	 24000
+		fir_passpand_ripple	 10
+		fir_stopband_ripple	 -95000
```

Comment:
@lgirdwood @juimonen what is the plan for DMIC support? Are we really going to generate a blob in the kernel based on this topology, or is the firmware going to generate the relevant register settings based on topology indications?